scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
+      * @param lengthFieldOffset the offset of the length field.
+      * @param lengthFieldLength  the length of the length field. Must be 1, 2, 3, 4 or 8 bytes.
+      * @param encodingLengthAdjustment the compensation value to add to the value of the length field when encoding.
+      * @param decodingLengthAdjustment the compensation value to add to the value of the length field when decoding.
+      * @param initialBytesToStrip the number of first bytes to strip out from the decoded frame. In standard framing setup
+      *                            it should be equal to lengthFieldLength.
+      * @param failFast  If true, a DecodingError event is generated as soon as the decoder notices the length of the frame will
+      *                  exceed maxFrameLength regardless of whether the entire frame has been read. If false,
+      *                  a a DecodingError event is generated after the entire frame that exceeds maxFrameLength has been read..
```

Comment:
```suggestion
      *                  a DecodingError event is generated after the entire frame that exceeds maxFrameLength has been read.
```
Why would you choose one or the other? What happens to the buffer if the error is raised fast?

Reply:
From what i know, in both cases buffer will discarded. In normal case it will be done after reading whole buffer, in `failFast` case it will be done as soon as violation is discvoered and after throwing `TooLongFrameException` decoder will enter byte discarding mode i.e it will discard next upcoming necessary bytes.

About choosing one or another, I am not sure. Main difference is that in fail fast mode downstream handlers will not get information about how big this violating frame would be. So there may be some protocols to react those violations.  ( we are not doing any of that thats why default is `failFast= true`)

Reply:
Yes we don't say anything in `DecodingError`. Right so Netty would drop as many bytes as the stream indicated, that makes sense, thanks!